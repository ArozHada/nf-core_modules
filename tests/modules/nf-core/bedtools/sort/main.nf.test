nextflow_process {

    name "Test Process BEDTOOLS_SORT"
    script "modules/nf-core/bedtools/sort/main.nf"
    process "BEDTOOLS_SORT"
    config "tests/modules/nf-core/bedtools/sort/nextflow.config"
    tag "bedtools_sort"

    test("Sarscov2 Genome BED") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [ [ id:'test'],
                            file(params.test_data['sarscov2']['genome']['test_bed'], checkIfExists: true)
                           ]
                input[1] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(path(params.outdir).list()).match("bed") }
            )
        }
    }

    test("Sarscov2 Genome BED With Genome") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [ [ id:'test'],
                            file(params.test_data['sarscov2']['genome']['test_bed'], checkIfExists: true)
                           ]
                input[1] = Channel.of(file(params.test_data['sarscov2']['genome']['genome_fasta_fai'], checkIfExists: true))
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(path(params.outdir).list()).match("bed_genome") }
            )
        }
    }
}
