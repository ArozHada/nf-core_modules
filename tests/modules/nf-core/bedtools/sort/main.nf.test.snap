{
    "bed": {
        "content": [
            [
                "test_out.testtext:md5,fe4053cf4de3aebbdfc3be2efb125a74",
                "versions.yml:md5,4d685e271db868cbbc9694abac17a7fe"
            ]
        ],
        "timestamp": "2023-04-11T22:22:09+0000"
    },
    "bed_genome": {
        "content": [
            [
                "test_out.testtext:md5,fe4053cf4de3aebbdfc3be2efb125a74",
                "versions.yml:md5,4d685e271db868cbbc9694abac17a7fe"
            ]
        ],
        "timestamp": "2023-04-11T22:22:09+0000"
    }
}